File: src/804. Unique Morse Code Words.js
/**
 * @param {string[]} words
 * @return {number}
 */

export function uniqueMorseRepresentations(words) {
  const morseCodes = [
      '.-',
      '-...',
      '-.-.',
      '-..',
      '.',
      '..-.',
      '--.',
      '....',
      '..',
      '.---',
      '-.-',
      '.-..',
      '--',
      '-.',
      '---',
      '.--.',
      '--.-',
      '.-.',
      '...',
      '-',
      '..-',
      '...-',
      '.--',
      '-..-',
      '-.--',
      '--..',
    ],
    transformations = new Set()
  for (let word of words) {
    let transformation = ''
    for (let char of word) {
      transformation += morseCodes[char.charCodeAt(0) - 97]
    }
    transformations.add(transformation)
  }
  return transformations.size
}

File: src/804. Unique Morse Code Words.cpp
class Solution {
    public:
        int uniqueMorseRepresentations(vector < string > & words) {
            vector < string > morseCodes = {
                '.-',
                '-...',
                '-.-.',
                '-..',
                '.',
                '..-.',
                '--.',
                '....',
                '..',
                '.---',
                '-.-',
                '.-..',
                '--',
                '-.',
                '---',
                '.--.',
                '--.-',
                '.-.',
                '...',
                '-',
                '..-',
                '...-',
                '.--',
                '-..-',
                '-.--',
                '--..'
            };
            set < string > transformations;
            for (string word: words) {
                string transformation;
                for (char character: word) {
                    transformation += morseCodes[character - 'a'];
                }
                transformations.insert(transformation);
            }
            return transformations.size();
        }
};